bpo-35758: Fix building on ARM + MSVC

gongminmin opened the pull request:
Msvc defines _M_ARM for arm target, but it doesn't have x87 control word. Need to disable it to prevent some compiling problems.

<!-- issue-number: [bpo-14802](https://bugs.python.org/issue14802) -->
https://bugs.python.org/issue14802
<!-- /issue-number -->

gongminmin committed:
Disable x87 control word for non-x86 targets

On msvc, x87 control word is only available on x86 target. Need to disable it for other targets to prevent compiling problems.

eamanu commented on Include/pyport.h:
```diff
@@ -406,7 +406,7 @@ extern "C" {
 
 /* get and set x87 control word for VisualStudio/x86 */
-#if defined(_MSC_VER) && !defined(_WIN64) /* x87 not supported in 64-bit */
+#if defined(_MSC_VER) && !defined(_WIN64) && !defined(_M_ARM) /* x87 not supported in 64-bit */
```
Seems good. But don't see activity in the issue on bugs.python.org of this change.

pitrou replied:
Should probably test `_M_IX86` instead (and drop the `_WIN64` check). See [predefined macros on MSVC](https://docs.microsoft.com/en-us/cpp/preprocessor/predefined-macros?view=vs-2017).

Also cc @zooba.

gongminmin replied:
Changed to check _M_IX86.

eamanu reviewed (changes requested):
You must sign CLA and add NEWS. Maybe this PR need more discuss. I recommend open a new bpo or send some comments to the issue to activate it.

blurb-it[bot] committed:
📜🤖 Added by blurb_it.

gongminmin committed:
Include immintrin.h on x86 and x64 only

Immintrin.h is only available on x86 and x64. Need to disable it for other targets to prevent compiling problems.

pitrou committed:
Update NEWS according to PR contents

pitrou committed:
Force CI checks to run
